app/controllers/cold_messages_controller.rb

```diff
@@ -1,63 +1,53 @@
 class ColdMessagesController < ApplicationController
   before_action :authenticate_user!
-  before_action :require_business!
-  before_action :require_new_conversation!
-  before_action :require_active_subscription!
+
+  rescue_from ColdMessage::BusinessBlank, with: :redirect_to_new_business
```

Comment:
using exceptions for control flow is a debated practice : sure it makes the code quite clearer and indeed makes this quite nice to read.

yet, in https://github.com/joemasilotti/railsdevs.com/pull/387/files#diff-1fffcf1a4effb60cbfb062145b6ececaffc723517e646b154034d46a0f447e46R12 you relied on a more object approach : return an object with easy to use status methods to check what happened.
why not use this approach here ? this could be less "magical", more explicit and focused to the places where it's actually needed

related :
- https://www.honeybadger.io/blog/benchmarking-exceptions-in-ruby-yep-theyre-slow/

Reply:
Agreed. Thanks for the push! I refactored to use a third state on `Result` and have the controller redirect dynamically. Overall I like this approach much better!